packages/vm/lib/evm/evm.ts

```diff
@@ -255,10 +255,18 @@ export default class EVM {
       totalGas = totalGas.add(returnFee)
     }
 
-    // if not enough gas
+    // Check for SpuriousDragon EIP-170 code size limit
+    let allowedCodeSize = true
+    if (
+      this._vm._common.gteHardfork('spuriousDragon') &&
+      result.returnValue.length > this._vm._common.param('vm', 'maxCodeSize')
+    ) {
+      allowedCodeSize = false
+    }
+    // If enough gas and allowed code size
     if (
       totalGas.lte(message.gasLimit) &&
-      (this._vm.allowUnlimitedContractSize || result.returnValue.length <= 24576)
+      (this._vm.allowUnlimitedContractSize || allowedCodeSize)
```

Comment:
`allowUnlimitedContractSize` seems to be never set. I think checking for `allowedCodeSize` is enough here? As the check if we are before Spurious Dragon is implicitly done above?

This would break this property though as it seems to be a VM option which users can manually set. Can we deprecate this option?

Reply:
Never mind I guess this is fine that we allow it.

Reply:
Yes, this was introduced here https://github.com/ethereumjs/ethereumjs-vm/pull/282 , it can be useful as a debugging option.